src/bun.js/api/bun/ms.zig

```diff
@@ -0,0 +1,308 @@
+/// Parse a time string like "2d", "1.5h", "5m" to milliseconds
+pub fn parse(input: []const u8) ?f64 {
+    if (input.len == 0 or input.len > 100) return null;
+
+    var i: usize = 0;
+
+    next: switch (input[i]) {
+        '-', '.', '0'...'9' => {
+            i += 1;
+            if (i < input.len) continue :next input[i];
+            break :next;
+        },
+        ' ', 'a'...'z', 'A'...'Z' => break :next,
+        else => return null,
+    }
```

Comment:
Also this behavior would be better organized in `immutable.zig` with a function, perhaps called `findFirstNonNumeric`.